### PR TITLE
#439: Set Maya defaults on file open

### PR DESCRIPTION
**Pull request by aparna-d2**
Modified hook scene_operation_tk-maya of tk-multi-workfiles2 app to set default render path, resolution, image format on file open.

### CODE REVIEW

File: hooks/tk-multi-workfiles2/scene_operation_tk-maya.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+            # get show resolution
+            indiapipeline_prefs = preferences.Preferences(package="indiapipeline")
+            try:
+                fields.update({"width": indiapipeline_prefs["show_resolution"]["width"],
+                               "height": indiapipeline_prefs["show_resolution"]["height"],
+                               "output": "LAYERPLACEHOLDER"})   # output is alphanumeric; replace with token <Layer>
+            except KeyError as ke:
+                # logging?
+                print "Unable to find {} in indiapipeline preferences. Not setting render defaults.".format(ke)
```
Comment (mbrainerd) [inline hooks/tk-multi-workfiles2/scene_operation_tk-maya.py]: Please use the SGTK logging module: self.parent.log_[info|warning|error|debug], or the newer self.parent.logger.[info|warning|error|debug]

File: hooks/tk-multi-workfiles2/scene_operation_tk-maya.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                    if not scene_name:
+                        cmds.SaveSceneAs()
+                    else:
+                        cmds.file(save=True)
+
+            # do new file:
+            cmds.file(newFile=True, force=True)
+            return True
+
+    def get_render_template(self, context):
```
Comment (mbrainerd) [inline hooks/tk-multi-workfiles2/scene_operation_tk-maya.py]: Missing docstring

File: hooks/tk-multi-workfiles2/scene_operation_tk-maya.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        elif entity_type == "Shot":
+            render_temp = templates["tk-maya_shot_work_render"]
+        elif entity_type == "Sequence":
+            render_temp = templates["tk-maya_sequence_work_render"]
+        else:  # elif type == "Project": (?)
+            render_temp = templates["tk-maya_project_work_render"]
+
+        return render_temp
+
+    def set_render_defaults(self, fields, render_path, frame_sq_key):
```
Comment (mbrainerd) [inline hooks/tk-multi-workfiles2/scene_operation_tk-maya.py]: Missing docstring

File: hooks/tk-multi-workfiles2/scene_operation_tk-maya.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+        self.unlock_and_set_attr("defaultRenderGlobals.imageFilePrefix", prefix, type="string", lock=True)
+
+        # ensure frame/animation ext = name.#.ext
+        self.set_enum_attr("defaultRenderGlobals.outFormatControl", "As Output Format", lock=True)
+        self.unlock_and_set_attr("defaultRenderGlobals.animation", True, lock=True)
+        self.unlock_and_set_attr("defaultRenderGlobals.putFrameBeforeExt", True, lock=True)
+        self.unlock_and_set_attr("defaultRenderGlobals.extensionPadding", int(frame_sq_key.format_spec), lock=True)
+        self.set_enum_attr("defaultRenderGlobals.periodInExt", "Period in Extension", lock=True)
+
+    def set_vray_settings(self, fields, render_path, frame_sq_key):
```
Comment (mbrainerd) [inline hooks/tk-multi-workfiles2/scene_operation_tk-maya.py]: Missing docstring

File: hooks/tk-multi-workfiles2/scene_operation_tk-maya.py [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+            self.unlock_and_set_attr("{}.imageFormatStr".format(node), ext, type="string", lock=True)
+            self.unlock_and_set_attr("{}.fileNamePrefix".format(node), prefix, type="string", lock=True)
+
+            # ensure frame/animation ext = name.#.ext
+            self.set_enum_attr("{}.animType".format(node), "Standard", lock=True)
+            self.unlock_and_set_attr("{}.fileNamePadding".format(node), int(frame_sq_key.format_spec), lock=True)
+
+    # move to a maya utils location?
+    def set_enum_attr(self, attribute_full, value, lock=False):
```
Comment (mbrainerd) [inline hooks/tk-multi-workfiles2/scene_operation_tk-maya.py]: Missing docstring

File: hooks/tk-multi-workfiles2/scene_operation_tk-maya.py [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+            if '=' in enum_item:
+                index = enum_item.split('=')[1]
+                enum_dict[enum_item.split('=')[0]] = index
+            else:
+                enum_dict[enum_item] = index
+            index += 1
+
+        self.unlock_and_set_attr(attribute_full, enum_dict[value], lock=lock)
+
+    def unlock_and_set_attr(self, attribute, value, type=None, lock=False):
```
Comment (mbrainerd) [inline hooks/tk-multi-workfiles2/scene_operation_tk-maya.py]: Missing docstring

File: hooks/tk-multi-workfiles2/scene_operation_tk-maya.py [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+        cmds.setAttr(attribute, lock=False)
+        if type:
+            cmds.setAttr(attribute, value, type=type, lock=lock)
+        else:
+            cmds.setAttr(attribute, value, lock=lock)
+
+    def split_prefix_ext(self, render_path, frame_sq_key):
```
Comment (mbrainerd) [inline hooks/tk-multi-workfiles2/scene_operation_tk-maya.py]: Missing docstring

File: hooks/tk-multi-workfiles2/scene_operation_tk-maya.py [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+            # do new file:
+            cmds.file(newFile=True, force=True)
+            return True
+
+    def get_render_template(self, context):
+        fields = context.as_template_fields()
+        templates = context.sgtk.templates
+        entity_type = context.entity["type"]
+
+        if entity_type == "Asset":
```
Comment (mbrainerd) [inline hooks/tk-multi-workfiles2/scene_operation_tk-maya.py]: Please replace this with:
```
template_exp = "{engine_name}_{env_name}_work_render"
template_name = self.parent.resolve_setting_expression(template_exp)
render_temp = templates[template_name]
```

File: hooks/tk-multi-workfiles2/scene_operation_tk-maya.py (hunk 1 above)
Comment (aparna-d2) [reply hooks/tk-multi-workfiles2/scene_operation_tk-maya.py]: Ah, forgot to do that.

File: hooks/tk-multi-workfiles2/scene_operation_tk-maya.py (hunk 8 above)
Comment (aparna-d2) [reply hooks/tk-multi-workfiles2/scene_operation_tk-maya.py]: Didn't realize you could do that. Thanks!

**Pull request merged**